src/javascript/app/pages/cashier/cashier.js

```diff
@@ -65,14 +69,86 @@ const Cashier = (() => {
         });
     };
 
+    const showCurrentCurrency = (currency, has_no_mt5, has_no_tx) => {
+        const el_acc_currency     = getElementById('account_currency');
+        const el_currency_image   = getElementById('account_currency_img');
+        const el_current_currency = getElementById('account_currency_current');
+        const el_current_hint     = getElementById('account_currency_hint');
+
+        const missingCriteria     = (has_mt5, has_tx) => {
+            const existing_mt5_msg = localize('You can no longer change the currency because you’ve created an MT5 account. [_1]Manage your accounts[_2].', [`<a href=${urlFor('user/accounts')}>`, '</a>']);
+            const existing_tx_msg  = localize('You can no longer change the currency because you’ve made a first-time deposit. [_1]Manage your accounts[_2].', [`<a href=${urlFor('user/accounts')}>`, '</a>']);
+
+            return !has_mt5 && has_tx ? existing_tx_msg : (has_mt5 && !has_tx ? existing_mt5_msg : existing_tx_msg);
+        };
+
+        // Set messages based on if currency is crypto or fiat
+        // If fiat, set message based on if they're allowed to change currency or not
+        // Condition is to have no MT5 accounts *and* have no transactions
+        const currency_message    = isCryptocurrency(currency)
+            ? localize('This is your [_1] account.', `${currency}`)
+            : has_no_mt5 && has_no_tx
+                ? localize('Your fiat account’s currency is currently set to [_1].', `${currency}`)
+                : localize('Your fiat account’s currency is set to [_1].', `${currency}`);
+
+        const currency_hint       = isCryptocurrency(currency)
+            ? localize('Don\'t want to trade in [_1]? You can open another cryptocurrency account. [_2]Manage your accounts[_3].', [`${currency}`, `<a href=${urlFor('user/accounts')}>`, '</a>'])
+            : has_no_mt5 && has_no_tx
+                ? localize('You can set a new currency before you deposit for the first time or create an MT5 account. [_1]Manage your accounts[_2].', [`<a href=${urlFor('user/accounts')}>`, '</a>'])
+                : missingCriteria(!has_no_mt5, !has_no_tx);
+
+        elementInnerHtml(el_current_currency, currency_message);
+        elementInnerHtml(el_current_hint, currency_hint);
+        el_currency_image.src = `/images/pages/cashier/icons/icon-${currency}.svg`;
```

Comment:
@msamprz you can use the currencies icon from `images/pages/set_currency` instead of adding the new ones (unless they are different), they looked the same to me 😄

Reply:
Due to inconsistency of the shape of icons of the new set vs the ones in `images/pages/set_currency`, this will be addressed by adding a wrapper to the image to be able to scale the image (so nothing is cropped) and have a circular window of it.

To avoid this:
![image](https://user-images.githubusercontent.com/15208816/55146359-60e7a100-517f-11e9-9b8c-c4f09a2c134b.png)

And look like this (only issue with this is that the image will be scaled):
![image](https://user-images.githubusercontent.com/15208816/55146422-84125080-517f-11e9-9504-451ffaf7e219.png)

Reply:
okay, let's keep these new icons then 👍